### PR TITLE
Update UserGuide.md

### LINKED ISSUE #139: Update UserGuide.md
Update manage contact features and any planned enhancements

### PR DESCRIPTION
**Pull request by limjunxian1**
This PR will:
1. Finish updating all features under manage contact except `updatephoto`
2. Updates notes under feature section and planned enhancements for some minor bugs that cannot be fixed (and people may complain that it is a bug)

Closes #139

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -170,16 +171,16 @@ All planned enhancements will also be listed in the [Planned Enhancements / Know
 **Notes about the command format:**<br>
 
 * Words in `UPPER_CASE` are the parameters to be supplied by the user.<br>
-  e.g. in `add n/NAME`, `NAME` is a parameter which can be used as `add n/John Doe`.
+  e.g. In `add n/NAME`, `NAME` is an parameters which can be used as `add n/John Doe`.
```
Comment (lordidiot) [inline docs/UserGuide.md]: Isn't `a parameter` correct here?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -170,16 +171,16 @@ All planned enhancements will also be listed in the [Planned Enhancements / Know
...
 
-* Items in square brackets are optional, ___with `addalt` as an exception. (Refer to [Add alternative information to contact](#add-alternative-information-to-contact) under Features)___<br>
-  e.g `n/NAME [t/TAG]` can be used as `n/John Doe t/friend` or as `n/John Doe`.
+* Parameters in square brackets are optional fields, **with `addalt` and `edit` as an exception. (Refer to [Add alternative information to contact](#add-alternative-information-to-contact) and [Edit contact information](#edit-contact-information) under Features)**<br>
```
Comment (lordidiot) [inline docs/UserGuide.md]:
```suggestion
* Parameters in square brackets are optional, **with `addalt` and `edit` as an exception. (Refer to [Add alternative information to contact](#add-alternative-information-to-contact) and [Edit contact information](#edit-contact-information) under Features)**<br>
```
Not all parameters are fields

File: docs/UserGuide.md [hunk 3]
```diff
@@ -170,16 +171,16 @@ All planned enhancements will also be listed in the [Planned Enhancements / Know
...
+  e.g In `n/NAME [t/TAG]`, `[t/TAG]` is an optional field which can be used as `n/John Doe t/friend` or as `n/John Doe`.
```
Comment (lordidiot) [inline docs/UserGuide.md]:
```suggestion
  e.g In `n/NAME [t/TAG]`, `[t/TAG]` is an optional parameter which can be used as `n/John Doe t/friend` or as `n/John Doe`.
```
ditto

File: docs/UserGuide.md [hunk 4]
```diff
@@ -189,91 +190,105 @@ All planned enhancements will also be listed in the [Planned Enhancements / Know
...
 </box>
 
-Examples:
-* `addalt 1 tg/@johndoe e2/johndoe@hotmail.com`
-    * Adds John Doe's telegram "@johndoe" and secondary email "johndoe@hotmail.com"
-* `addalt 1 li/john-doe-b9a38128a`
-    * Adds John Doe's linkedin "john-doe-b9a38128a"
+<box type="warning">
+
+* You should provide a positive integer that is smaller than the number of contacts stored in CampusConnect for `PERSON_INDEX`.
```
Comment (lordidiot) [inline docs/UserGuide.md]: Should it be smaller than the number of contacts currently listed?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -189,91 +190,105 @@ All planned enhancements will also be listed in the [Planned Enhancements / Know
...
+* `edit 2 n/Betsy e/betsy@example.com a/Newgate Heavan p/98765411 t/bestfriend`: Edits name to "Besty", email to "besty@example.com", address to "Newgate Heavan", phone to "98765411" and tag to "bestfriend" for contact at index 2 of your contact list.  
+
+Refer to [properties of contact](#properties-of-contact) on the accepted formats for the respective parameters.
+
+</box>
 
 <box type="warning">
 
-You are not allowed to edit any alternative contact information, i.e. `TELEGRAM`, `SECONDARY_EMAIL`, `LINKEDIN`, `BIRTHDAY` if any of these fields are empty. You will receive an error message that directs you to use [**`addalt`**](#add-alternative-information-to-contact) command.
+* You should provide a positive integer that is smaller than the number of contacts stored in CampusConnect for `PERSON_INDEX`.
```
Comment (lordidiot) [inline docs/UserGuide.md]: ditto

File: docs/UserGuide.md [hunk 6]
```diff
@@ -295,21 +310,41 @@ You are required to input the absolute path to the photo for this command. If yo
 
 ###  List all contacts: `list`
 
-Shows a list of all contacts.
+After using CampusConnect's [find contact](#find-contacts) feature that filters the contacts you are looking for, you may want to have an oversight of all your contacts again! Thus, CampusConnect allows you to list all your contacts.
```
Comment (lordidiot) [inline docs/UserGuide.md]:
```suggestion
After using CampusConnect's [find contact](#find-contacts) feature that filters the contacts you are looking for, you may want to have an overview of all your contacts again! Thus, CampusConnect allows you to list all your contacts.
```
I think `oversight` is the wrong word choice here

File: docs/UserGuide.md [hunk 7]
```diff
@@ -295,21 +310,41 @@ You are required to input the absolute path to the photo for this command. If yo
...
+
+Below are some examples on how to use the command:
+* `delete 1`: Deletes contact at index 1 of your contact list.
+* `delete 2`: Deletes contact at index 2 of your contact list.
+
+</box>
+
+<box type="warning">
+
+You should provide a positive integer that is smaller than the number of contacts stored in CampusConnect for `PERSON_INDEX`.
```
Comment (lordidiot) [inline docs/UserGuide.md]: ditto

File: docs/UserGuide.md [hunk 8]
```diff
@@ -544,16 +579,22 @@ Note that the last example is **not equivalent** to `n/do && t/friend || t/colle
 # Planned Enhancements / Known Issues
 
 1. **When using multiple screens**, if you move the application to a secondary screen, and later switch to using only the primary screen, the GUI will open off-screen. The remedy is to delete the `preferences.json` file created by the application before running the application again.
+2. **When executing `addalt` command**, if you input other prefixes that is not accepted by the command format, the error message shown does not prompt you to remove those prefixes and adhere strictly to the command format. We will be working on this in the future to improve the specificity of error messages.
```
Comment (lordidiot) [inline docs/UserGuide.md]:
```suggestion
2. **When executing `addalt` command**, if you input other prefixes that are not accepted by the command format, the error message shown does not prompt you to remove those prefixes and adhere strictly to the command format. We will be working on this in the future to improve the specificity of error messages.
```

File: docs/UserGuide.md [hunk 9]
```diff
@@ -544,16 +579,22 @@ Note that the last example is **not equivalent** to `n/do && t/friend || t/colle
...
+3. **When executing `edit` command**, if you try to edit a specified contact to have the same [properties](#properties-of-contact) of another saved contact **(besides `NAME` with the same casing) in your contact list, CampusConnect allows you to do so. We plan to enhance the `edit` command such that it takes into account what makes a contact unique in your contact list.
```
Comment (lordidiot) [inline docs/UserGuide.md]: Wow nice catch

**Review by lordidiot — CHANGES_REQUESTED** [review]
Looks really good. Just some minor language changes, then let's merge this 👍

File: docs/UserGuide.md (hunk 1 above)
Comment (limjunxian1) [reply docs/UserGuide.md]: yes agreed, fixed in b33ffa82

File: docs/UserGuide.md (hunk 2 above)
Comment (limjunxian1) [reply docs/UserGuide.md]: fixed in b33ffa82

File: docs/UserGuide.md (hunk 3 above)
Comment (limjunxian1) [reply docs/UserGuide.md]: fixed in b33ffa82

File: docs/UserGuide.md (hunk 4 above)
Comment (limjunxian1) [reply docs/UserGuide.md]: nice catch, the index definitely could be the same as the number of contacts! fixed in b33ffa82

File: docs/UserGuide.md (hunk 5 above)
Comment (limjunxian1) [reply docs/UserGuide.md]: fixed in b33ffa82

File: docs/UserGuide.md (hunk 6 above)
Comment (limjunxian1) [reply docs/UserGuide.md]: fixed in b33ffa82

File: docs/UserGuide.md (hunk 7 above)
Comment (limjunxian1) [reply docs/UserGuide.md]: fixed inn b33ffa82

File: docs/UserGuide.md (hunk 8 above)
Comment (limjunxian1) [reply docs/UserGuide.md]: nice catch of grammatical errors, fixed in b33ffa82

**Comment by limjunxian1** [comment]
realised that there is a misinterpretation of the comment on `PERSON_INDEX` instruction under each warning box, have fixed in 37e72e5

**Review by lordidiot — APPROVED** [review]
LGTM! Thank you for working on all the changes, very thorough work 👍

**Pull request merged**